patches/tModLoader/Terraria/DataStructures/PlayerDeathReason.cs.patch

```diff
@@ -40,6 +42,7 @@
 +	public ref int SourceProjectileType => ref _sourceProjectileType;
 +	public ref Item SourceItem => ref _sourceItem;
 +	public ref string SourceCustomReason => ref _sourceCustomReason;
```

Comment:
Also obsolete this one.

Reply:
Ideally, we could convert it to a wrapper property which does `NetworkText.FromLiteral` and `.ToString()` on `set/get` but... up to you whether we want to introduce runtime breakage.

Reply:
I think I'll avoid breakage, the obsolete should be all that is needed for modders to migrate to the new approach if that is what they are working with.